actionpack/lib/action_dispatch/middleware/exception_wrapper.rb

```diff
@@ -1,5 +1,11 @@
 # frozen_string_literal: true
 
+begin
+  # Activate error_highlight 0.4.0 or later if possible
+  gem "error_highlight", ">= 0.4.0"
+rescue LoadError
+end
```

Comment:
Sorry I wasn't clear. We should record in a variable or something whether or not we managed to load the right version. So that we can later use that in conditionals.

Also I think `gem` doesn't load, so the full pattern may need to be:

```ruby
has_error_highlight = begin
  gem ...
  require ...
  true
rescue LoadError
  false
end

Reply:
Also note to myself, ExceptionWrapper is supposed to only be loaded in test, but several production code paths reference it, I should refactor that.

Reply:
Oh sorry, the approach doesn't work. The interpreter requires `error_highlight` at the startup, so error_highlight 0.3.0 is activated if Gemfile has no explicit denendency.

```
$ cat Gemfile
source "https://rubygems.org"

$ cat foo.rb
p ErrorHighlight::VERSION

$ bundle install

$ bundle exec ruby foo.rb
"0.3.0"
```

The idiom `gem "error_highlight", ">= 0.4.0"` does not work as error_highlight 0.3.0 is already activated.

```
$ cat foo.rb
gem "error_highlight", ">= 0.4.0"
p ErrorHighlight::VERSION

$ bundle exec ruby foo.rb
/.../rubygems_integration.rb:280:in `block (2 levels) in replace_gem': can't activate error_highlight (>= 0.4.0), already activated error_highlight-0.3.0. Make sure all dependencies are added to Gemfile. (Gem::LoadError)
        from foo.rb:1:in `<main>'
```

I guess there is no other way than to put `gem "error_highlight", ">= 0.4.0"` in Gemfile (or use Ruby 3.2 which will bundle error_highlight 0.4.0 or later). Then the interpreter activates 0.4.0 at the startup time, so everything works well.


```
$ cat Gemfile
source "https://rubygems.org"
gem "error_highlight", ">= 0.4.0"

$ cat foo.rb
p ErrorHighlight::VERSION

$ bundle install

$ bundle exec ruby foo.rb
"0.4.0"
```

Reply:
Ah that's unfortunate :/

Well, then just checking version should be fine then.

Reply:
I'll delete the idiom at the top of the file for now. But thanks for your suggestion!